src/server.rs

```diff
@@ -219,11 +219,19 @@ fn recv_loop(peer_id: usize, stream: &TcpStream, server_tx: Sender<Event>) -> Re
     let msg = Message::New(stream.try_clone()?);
     server_tx.send(Event { peer_id, msg })?;
 
+    let mut first_line = true;
     for line in BufReader::new(stream).lines() {
+        if let Err(ref e) = line {
+            warn!("{}: recv error: {:?} - {}", peer_id, e.kind(), e);
```

Comment:
Isn't this logged already?

Reply:
The only thing emitted when connecting with an SSL client without this change is:
```
 [2021-12-01T20:40:47.681Z WARN  electrs::thread] recv_loop thread failed: 2: recv failed
```
Knowing the error kind and string can be helpful.

Reply:
Hmm, would be a bit nicer to fix error reporting done by a caller.

Reply:
Yeah, I agree that having `spawn()` in *thread.rs* unwrap the `anyhow` error and append information about the inner error. I'll see if I can figure out how to do that.

Reply:
Alright, 844d7c2 results in the following output:
```
[2021-12-01T21:58:36.290Z WARN  electrs::server] InvalidData on first line may indicate client attempted to connect using SSL when server expects unencrypted communication.
[2021-12-01T21:58:36.290Z WARN  electrs::thread] recv_loop thread failed: 0: recv failed
[2021-12-01T21:58:36.290Z WARN  electrs::thread] because: stream did not contain valid UTF-8
```